crates/net/eth-wire/src/types/broadcast.rs

```diff
@@ -378,26 +395,106 @@ impl Decodable for NewPooledTransactionHashes68 {
+            return Err(alloy_rlp::Error::Custom(ValidationError68::EMPTY))
+        }
 
-        if msg.hashes.len() != msg.types.len() {
+        // 2. checks that indices of types, sizes and hashes lists map 1-1
+        if hashes.len() != types.len() {
             return Err(alloy_rlp::Error::ListLengthMismatch {
-                expected: msg.hashes.len(),
-                got: msg.types.len(),
+                expected: hashes.len(),
+                got: types.len(),
             })
         }
-        if msg.hashes.len() != msg.sizes.len() {
+        if hashes.len() != sizes.len() {
             return Err(alloy_rlp::Error::ListLengthMismatch {
-                expected: msg.hashes.len(),
-                got: msg.sizes.len(),
+                expected: hashes.len(),
+                got: sizes.len(),
             })
         }
 
-        Ok(msg)
+        // 3. checks values are valid
+        //
+        // (i) checks if hashes values are valid
+        for (i, &hash) in hashes.iter().enumerate() {
+            debug!(target: "net::eth-wire",
+                ty=types[i],
+                size=sizes[i],
+                hash=%hash,
+                err=ValidationError68::INVALID_TX_HASH,
+                "invalid eth68 announcement"
+            );
+            if hash == TxHash::ZERO {
+                return Err(alloy_rlp::Error::Custom(ValidationError68::INVALID_TX_HASH))
+            }
+        }
+        // (ii) checks if tx types are valid values, we don't really care if the tx type is wrong
+        // for an invalid hash, so we do this second
+        for (i, &ty) in types.iter().enumerate() {
+            debug!(target: "net::eth-wire",
+                ty=ty,
+                size=sizes[i],
+                hash=%hashes[i],
+                err=ValidationError68::INVALID_TX_TYPE,
+                "invalid eth68 announcement"
+            );
+            if TxType::try_from(ty).is_err() {
+                return Err(alloy_rlp::Error::Custom(ValidationError68::INVALID_TX_TYPE))
+            }
+        }
+        // (iii) checks if sizes are valid values, we don't really care if the size is wrong for a
+        // tx type that doesn't follow schema, so we do this third
+        for (i, &size) in sizes.iter().enumerate() {
+            if size > TX_MAX_SIZE || size == 0 {
+                // todo: min tx size?
+                debug!(target: "net::eth-wire",
+                    ty=types[i],
+                    size=size,
+                    hash=%hashes[i],
+                    TX_MAX_SIZE=TX_MAX_SIZE,
+                    err=ValidationError68::INVALID_TX_SIZE,
+                    "invalid eth68 announcement"
+                );
+
+                return Err(alloy_rlp::Error::Custom(ValidationError68::INVALID_TX_SIZE))
+            }
+        }
+
+        // 4. checks for duplicate hash values
+        if hashes.len() > 1 {
+            let mut hashes_clone: SmallVec<[TxHash; 6]> = hashes.clone().into();
```

Comment:
@DaniPopes would you agree 6 x 32 byte hashes are fine to push on the stack here?

Reply:
probably fine if we expect the majority of cases to contain <=6 and this is perf-sensitive. You want `SmallVec::copy_from_slice` here instead of `.clone().into()`